src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityPage.java

```diff
@@ -0,0 +1,162 @@
+
+    private WebElementAction webElementAction = new WebElementAction();
+
+    @FindBy(xpath = "//div/textarea[@class=\" textarea\"]")
+    private WebElement descriptionTxtBox;
+
+    private By saveBtnXpath = By.xpath("//button[@title='" + translateValue("LegalEntities", "button.save") + "']");
+    private By streetTxtBoxXpath = By.xpath("//div/textarea[@placeholder='"
+            + translateValue("LegalEntities", "input.street") + "']");
+    private static final String INTERNATIONALIZED_NAME = translateValue("LegalEntities", "input.name");
+    private static final String INTERNATIONALIZED_COMPANY_NAME =
+            translateValue("LegalEntities", "input.companyname");
+    private static final String INPUT_XPATH = "//label/span[text()='%s']/../..//input";
+    private static final String INPUT_ADDRESS_CSS = "input.%s";
+    private static final String DROPDOWN_XPATH = "//a[@class='%s']";
+    private static final String DROPDOWN_OPTION_XPATH = "//div[@class=\"select-options\"]//a[@title='%s']";
+    private static final HashMap<String, String> INPUT_FIELDS_NAMES = new HashMap<>();
+
+    static {
+        INPUT_FIELDS_NAMES.put("Name", INTERNATIONALIZED_NAME);
+        INPUT_FIELDS_NAMES.put("CompanyName", INTERNATIONALIZED_COMPANY_NAME);
+
+    }
+
+    private static final HashMap<String, String> INPUT_ADDRESS_NAMES = new HashMap<>();
+
+    static {
+        INPUT_ADDRESS_NAMES.put("City", "city");
+        INPUT_ADDRESS_NAMES.put("State", "state");
+        INPUT_ADDRESS_NAMES.put("PostalCode", "postalCode");
+        INPUT_ADDRESS_NAMES.put("Country", "country");
+    }
+
+    /**
+     * Waits for the visibility of a web element.
+     */
+    @Override
+    protected void waitForPageToLoad() {
+        getWait().until(ExpectedConditions.visibilityOf(descriptionTxtBox));
+    }
+
+    /**
+     * Introduces text to web email text box.
+     *
+     * @param fieldName the field name of the map.
+     * @param value a String to input
+     */
+    public void setInputFieldWithInternationalization(final String fieldName, final String value) {
+        webElementAction.setTextField(webElementAction
+                .getWebElementByXpathAndValue(INPUT_XPATH, INPUT_FIELDS_NAMES.get(fieldName)), value);
+    }
+
+    /**
+     * Introduces text to web email text box.
+     *
+     * @param fieldName the field name of the map.
+     * @param value a String to input
+     */
+    public void setInputFieldByClass(final String fieldName, final String value) {
+        webElementAction.setTextField(webElementAction
+                .getWebElementByCssAndValue(INPUT_ADDRESS_CSS, INPUT_ADDRESS_NAMES.get(fieldName)), value);
+    }
+
+    /**
+     * Selects an option from a dropdown menu.
+     *
+     * @param fieldName the name of the menu field.
+     * @param option    the option to select.
+     * @return a NewLegalEntity.
+     */
+    public NewLegalEntityPage selectFromDropDown(final String fieldName, final String option) {
```

Comment:
Not really sure. Why some methods return a NewLegalEntityPage and some are void? Is it because of the feature?

Reply:
It was my mistake, thank you for the observation, I'll fix it.